kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/storage/ProducerStateManager.java

```diff
@@ -196,15 +203,44 @@ public void completeTxn(CompletedTxn completedTxn) {
         }
     }
 
+    public boolean hasSomeAbortedTransactions() {
+        return !abortedIndexList.isEmpty();
+    }
+
+    public void purgeAbortedTxns(long offset) {
+        boolean empty;
+        boolean somethingDone;
+        synchronized (abortedIndexList) {
+            somethingDone = abortedIndexList.removeIf(tx -> {
```

Comment:
Since the aborted txn list could be sorted by offset, is it worth optimizing this to order the list and then stop iterating at the first aborted txn that we can keep?

Reply:
makes sense, the list may become big